### PR TITLE
Fix bugs related to fit limits in GSAS2 tab in EngDiff UI

### LINKED ISSUE #35597: Error in GSAS2 tab of EngDiff UI when adjusting fitting range
**Describe the bug**
Found during manual testing #35541
Error in GSAS2 tab of EngDiff UI when adjusting fitting range

```
Traceback (most recent call last):  File "/opt/mantidworkbenchnightly/lib/python3.8/site-packages/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/presenter.py", line 36, in on_refine_clicked 
number_output_histograms = self.model.run_model(  File "/opt/mantidworkbenchnightly/lib/python3.8/site-packages/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/model.py", line 115, in run_model 
self.load_basic_outputs(gsas_result)  File "/opt/mantidworkbenchnightly/lib/python3.8/site-packages/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/model.py", line 692, in load_basic_outputs
self.create_lattice_parameter_table()  File "/opt/mantidworkbenchnightly/lib/python3.8/site-packages/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/model.py", line 810, in create_lattice_parameter_table
with open(parameters_txt, "rt", encoding="utf-8") as file:FileNotFoundError: [Errno 2] No such file or directory: '/home/ra1112751/Engineering_Mantid/GSAS2/hello/hello_cell_parameters_Fe_gamma.txt'
```

**To Reproduce**
Test 11 of manual testing instructions
(https://developer.mantidproject.org/Testing/EngineeringDiffraction/EngineeringDiffractionTestGuide.html)

**Expected behavior**
Would pass fitting range to GSAS2 refine command and would retrieve results without error

**Screenshots**

**Platform/Version (please complete the following information):**
 - OS: [e.g. Windows, RHEL 7, Ubuntu, macOS]
 - OS Version:
 - Mantid Version [e.g. 6.0.0]

**Additional context**

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
**Description of work**
Fixes 2 bugs (not regressions - hence the release notes -  but it would be nice to get into 6.7 as they are easy to trigger):

1. Checks reflections file exists before trying to read it (this happens if no reflections in TOF limits - but there may be other reasons - e.g. if refinement fails?)
2. Stop users entering min > max using sliders or edit boxes (atm it doesn't do anything fancy it just resets the slider and box to the data min/max)

Long term it would be good to only generate reflections within limits of the data/sliders (also the max d-spacing is hard-coded to 3 Ang)
https://github.com/mantidproject/mantid/blob/df7bfd91c8df842f0327afdc06b4756cf3eca05a/qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/model.py#L455

Also I think the slider code could be made a bit more concise and tidied up a  bit - but that can wait till after the release.

**To test:**

Follow instructions on original issue.

After refining data in GSAS-II the data and fit will be plotted along with sliders for the fitting range. 
1. Move the sliders to check min < max (slider position reset to data min.max) and check the edit box is correct
2. Check that that trying to set min > max in the edit boxes will reset the slider position to data min/max

Fixes #35597 

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/call_G2sc.py [hunk 1]
```diff
@@ -90,10 +90,8 @@ def enable_unit_cell(refine, override_unit_cell_lengths, project):
 
 def enable_limits(x_limits, project):
```
Comment (AnthonyLim23) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/call_G2sc.py:91]: is there a test for this?

**Comment by RichardWaiteSTFC** [comment]
Added maintenance issue to add tests for view and code that calls GSAS 2 #35624

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/call_G2sc.py (hunk 1 above)
Comment (RichardWaiteSTFC) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/call_G2sc.py:91]: As discussed offline (but commenting here for anyone else interested) only the model and presenter have tests

**Comment by AnthonyLim23** [comment]
I can update the fitting ranges in the GSAS tab by dragging the dashed line and by editing the combo box. 

I would suggest updating the docs to have the full paths (below) for the files. This was the only way I could get it to work.
```
//isis/inst$/NDXENGINX/Instrument/data/cycle_19_1/ENGINX00305738.nxs
//isis/inst$/NDXENGINX/Instrument/data/cycle_19_1/ENGINX00305761.nxs
//isis/inst$/NDXENGINX/Instrument/data/cycle_19_1/ENGINX00307521.nxs
```

**Comment by RichardWaiteSTFC** [comment]
> I can update the fitting ranges in the GSAS tab by dragging the dashed line and by editing the combo box. 


Thanks, yes that image looks like the fix has worked.

> I would suggest updating the docs to have the full paths (below) for the files. This was the only way I could get it to work.
> 
> ```
> //isis/inst$/NDXENGINX/Instrument/data/cycle_19_1/ENGINX00305738.nxs
> //isis/inst$/NDXENGINX/Instrument/data/cycle_19_1/ENGINX00305761.nxs
> //isis/inst$/NDXENGINX/Instrument/data/cycle_19_1/ENGINX00307521.nxs
> ```

OK, yes the archive search should work but I have come across this myself so I will update the manual testing instructions. If it's OK with you I would like to do that in a separate PR as I have to make some other changes as part of #35598 (corrections post manual testing)

**Review by AnthonyLim23 — APPROVED** [review]
Looks good  and works as expected

**Pull request merged**